src/Hl7.Fhir.Base/FhirPath/Parser/Grammar.cs

```diff
@@ -165,36 +200,55 @@ public static Parser<Expression> Invocation(Expression focus)
         // '.' invocation
         public static Parser<Expression> DotInvocation(Expression focus)
         {
-            return Parse.Char('.').Select(v => new SubToken(v)).Positioned().Then(op => FunctionInvocation(focus).Select(t => { t.SetPositionFrom(op.Location); return t; }));
+            return WhitespaceOrComments().Then(wsBeforeDot => Parse.Char('.').Select(v => new SubToken(v).WithLeadingWS(wsBeforeDot)).Positioned())
+                .Then(op => FunctionInvocation(focus).Select(t => { t.WithLeadingWS(op.LeadingWhitespace); return t; }));
         }
 
         // '[' expression ']'                             #indexerExpression
         public static Parser<Expression> IndexerInvocation(Expression focus)
         {
-            return Parse.Contained(Expression, Parse.Char('[').TokenIgnoreComments(), Parse.Char(']').TokenIgnoreComments())
-                .Select(ix => new IndexerExpression(focus, ix)).Positioned();
+            return
+                // The whitespace before the '[' is captured in the ws1 token
+                // so that the position information for the indexer isn't confused with the comment
+                from wsLeading in WhitespaceOrComments()
+                from indexer in (
+                    from lparen in Parse.Char('[').Select(v => new SubToken(v)).Positioned()
```

Comment:
Where is the equivalent of the `TokenIgnoreComments` on the left hand side? So, where are ww capturing the whitespace between the `[` and the following expression?  Or is Expression capturing those?

Reply:
Expression captures those - just as it does in function arguments between the commas.
There is a unit test that covers this specific case too.
https://github.com/FirelyTeam/firely-net-sdk/blob/6046024baf75120465423899e9c343ede45c4e97/src/Hl7.FhirPath.Tests/Tests/FhirPathGrammarTest.cs#L372